server/src/main/java/org/opensearch/search/SearchHits.java

```diff
@@ -100,6 +107,39 @@ public SearchHits(
         this.sortFields = sortFields;
         this.collapseField = collapseField;
         this.collapseValues = collapseValues;
+        if (FeatureFlags.isEnabled(FeatureFlags.PROTOBUF_SETTING)) {
+            List<FetchSearchResultProto.SearchHit> searchHitList = new ArrayList<>();
+            for (int i = 0; i < hits.length; i++) {
+                FetchSearchResultProto.SearchHit.Builder searchHitBuilder = FetchSearchResultProto.SearchHit.newBuilder();
+                if (hits[i].getIndex() != null) {
+                    searchHitBuilder.setIndex(hits[i].getIndex());
+                }
+                searchHitBuilder.setId(hits[i].getId());
+                searchHitBuilder.setScore(hits[i].getScore());
+                searchHitBuilder.setSeqNo(hits[i].getSeqNo());
+                searchHitBuilder.setPrimaryTerm(hits[i].getPrimaryTerm());
+                searchHitBuilder.setVersion(hits[i].getVersion());
+                if (hits[i].getSourceRef() != null) {
+                    searchHitBuilder.setSource(ByteString.copyFrom(hits[i].getSourceRef().toBytesRef().bytes));
+                }
+                searchHitList.add(searchHitBuilder.build());
+            }
+            QuerySearchResultProto.TotalHits.Builder totalHitsBuilder = QuerySearchResultProto.TotalHits.newBuilder();
+            totalHitsBuilder.setValue(totalHits.value);
+            totalHitsBuilder.setRelation(
+                totalHits.relation == Relation.EQUAL_TO
+                    ? QuerySearchResultProto.TotalHits.Relation.EQUAL_TO
+                    : QuerySearchResultProto.TotalHits.Relation.GREATER_THAN_OR_EQUAL_TO
+            );
+            FetchSearchResultProto.SearchHits.Builder searchHitsBuilder = FetchSearchResultProto.SearchHits.newBuilder();
+            searchHitsBuilder.setMaxScore(maxScore);
+            searchHitsBuilder.addAllHits(searchHitList);
+            searchHitsBuilder.setTotalHits(totalHitsBuilder.build());
+            if (collapseField != null) {
+                searchHitsBuilder.setCollapseField(collapseField);
+            }
+            this.searchHitsProto = searchHitsBuilder.build();
+        }
```

Comment:
Stub this out to `hits` to `HitsProto` method ?

Reply:
Done.